### ISSUE TITLE
Multicast requests using blockwise currently not working

### ISSUE THREAD

**Issue by JKRhb** [shamblett/coap#100]
In the current version of the library, using multicast with block-wise transfer currently does not work anymore, as the client checks if the response's token matches the one of the original request.

https://github.com/shamblett/coap/blob/3f03f45e66ca4102ce729a95a0477fab4acd1292/lib/src/coap_client.dart#L474

However, because the client switches to unicast when using block-wise transfer with a server, this check always returns `false` because a new token is assigned once the switch is made. I will try to come up with a fix for this problem in the next couple of days.

Maybe the check could simply return `true` if multicast is being used, like so?

```dart
.where((final e) => req.multicast || e.resp.token!.equals(req.token!))
```

Or could we add a field for an "original token" somewhere, which could be used for matching instead?

**Comment by JosefWN** [comment]
Ah, ouch. The token for the "overarching" request is set in the exchange when `fireRespond` is called: https://github.com/shamblett/coap/blob/4.2.1/lib/src/net/coap_exchange.dart#L154

For blockwise this is the last token in the blockwise exchange.

I can test a bit too. Just for reference, server that could be used for testing (can be modified to return blockwise): https://github.com/plgd-dev/go-coap/tree/master/examples/mcast/server

Client:
```dart
FutureOr<void> main(final List<String> args) async {
  final conf = CoapConfig();
  final uri = Uri(
    scheme: 'coap',
    host: CoapDefinedAddress.allCOAPNodesIPV4,
    port: conf.defaultPort,
  );
  final client = CoapClient(uri, conf);

  try {
    final req = CoapRequest.newGet()..uriPath = '/oic/res';
    await client.send(
      req,
      onMulticastResponse: CoapMulticastResponseHandler(print),
    );
  } on Exception catch (e) {
    print('CoAP encountered an exception: $e');
  }

  client.close();
}
```

**Comment by JosefWN** [comment]
Ah, my bad, I was printing all of the events and must have mistaken `CoapRespondingEvent` for `CoapRespondEvent`. Will dig a bit more, it could have to do with how the exchange is duplicated.

**Comment by JosefWN** [comment]
Yeah, tried and it works in `v3.5.0`, so it was my refactor.

An attempt at a fix with a multicast token like you were thinking of: https://github.com/shamblett/coap/pull/102

**Comment by JosefWN** [comment]
We can keep this open until this is fixed:
> Thank you, @JosefWN, block-wise multicast requests now work again :) The only thing I noticed is that the client seems to hang if you use multicast even if you explicitly close it. Unfortunately, I haven't been able to find the point where to fix the issue yet. Destroying the eventBus in the client, for example, does not have an effect on the behavior.

Investigating...

**Comment by shamblett** [comment]
Believe this has been addressed. Package re released at version 5.0.0.
